### PR TITLE
Integrating Caked to Affinity

### PR DESCRIPTION
**Pull request by crangelsmith**
[Caked]( https://github.com/alan-turing-institute/caked/tree/main) is being developed to allow for a more flexible data loader. It should do everything that is already implemented in affinity and more (e.g integration with CZI portal, parakeet, etc). 

This PR, integrates Caked with affinity, making the `data.py` a lot lighter in code. 

Commit history is a mess due to me trying to keep up to date with `develop`, but the files changed are the right ones to review.

### CODE REVIEW

**Review by marjanfamili — COMMENTED** [review]
Would it be possible to have a docstring in the overload load_dat() functions to explain why they are necessary? I am assuming is it for calling in evaluation mode ?

File: avae/data.py [hunk 1]
```diff
@@ -123,209 +118,209 @@ def load_data(
...
 
-    def __getitem__(self, item):
-        filename = self.paths[item]
+    def __getitem__(self, index):
 
-        data = np.array(self.read(filename))
-        x = self.voxel_transformation(data)
+        # read data from path
+        data = np.array(self.read(self.paths[index]))
+        x = self.transformation(data)
```
Comment (marjanfamili) [inline avae/data.py:321]: what is self.transformation() ?

**Review by marjanfamili — APPROVED** [review]
The code runs fine, however I find all the new additions slightly confusing, specially because one has to switch back and force between caked and affinity to see what all the functions are. it would be good to have some short comments to explain what is happening when they are being used. 


I have tested this on Mac for Mnist data with few transformations. The installation of caked works well in existing python environment without  any conflict.

File: avae/data.py (hunk 1 above)
Comment (crangelsmith) [reply avae/data.py:321]: this is the equivalent to voxel_transformation, but as we should be able to deal with very different datasets (1D, 2D and 3D) it was more clear to leave it a `transformation` only.

**Comment by crangelsmith** [comment]
> Would it be possible to have a docstring in the overload load_dat() functions in data.py to explain why they are necessary? I am assuming is it for calling in evaluation mode ?

Done in #[88bbd47](https://github.com/alan-turing-institute/affinity-vae/pull/293/commits/88bbd47e6da64113d8bad3ad748b51acd0e8c74b)

**Comment by crangelsmith** [comment]
> The code runs fine, however I find all the new additions slightly confusing, specially because one has to switch back and force between caked and affinity to see what all the functions are. it would be good to have some short comments to explain what is happening when they are being used.
> 
> I have tested this on Mac for Mnist data with few transformations. The installation of caked works well in existing python environment without any conflict.

Added more comments for context in [88bbd47](https://github.com/alan-turing-institute/affinity-vae/pull/293/commits/88bbd47e6da64113d8bad3ad748b51acd0e8c74b)

**Pull request merged**